packages/api/src/routes/medical/internal-patient.ts

```diff
@@ -353,4 +308,88 @@ router.post(
   })
 );
 
+const initEnhancedCoverageSchema = z.object({
+  cxId: uuiSchema.optional(),
+  fromOrgPos: stringIntegerSchema.optional(),
+});
+
+/** ---------------------------------------------------------------------------
+ * POST /internal/patient/enhance-coverage
+ *
+ * Trigger the job to enhance coverage of provided patients. Before doing that,
+ * it also checks/fixes any stale enhanced coverage process.
+ *
+ * @param req.query.cxId The customer ID (optional, default to all cxs with the
+ *                       respective Feature Flag enabled).
+ * @param req.query.fromOrgPos The position on the array of CQ Orgs to start the
+ *                             Enhanced Coverage from. If set, it disables the
+ *                             validation/check of Patient's cqLinkStatus
+ * @return 200 OK - processing asynchronously
+ */
+router.post(
+  "/enhance-coverage",
+  asyncHandler(async (req: Request, res: Response) => {
+    const { cxId, fromOrgPos } = initEnhancedCoverageSchema.parse(req.query);
+
+    const cxIds: string[] = cxId ? [cxId] : [];
+    if (!cxIds.length) {
+      cxIds.push(...(await getCxsWithEnhancedCoverageFeatureFlagValue()));
+    }
+
+    const checkStaleEC = !fromOrgPos || fromOrgPos <= 0;
+    if (checkStaleEC) await checkStaleEnhancedCoverage(cxIds);
+
+    const patientIds = await initEnhancedCoverage(cxIds, fromOrgPos);
+
+    return res.status(status.OK).json({ patientIds });
+  })
+);
+
+const cqLinkStatusSchema = z.enum(cqLinkStatus);
+
+const completeEnhancedCoverageSchema = z.object({
+  cxId: uuiSchema,
+  patientIds: uuiSchema.array(),
+  cqLinkStatus: cqLinkStatusSchema,
+});
+
+/** ---------------------------------------------------------------------------
+ * POST /internal/patient/enhance-coverage/completed
+ *
+ * Indicate the coverage enhancement has been completed.
+ *
+ * @param req.query.cxId The customer ID.
+ * @param req.body.patientIds The IDs of patient to update their CQLinkStatus and trigger doc query.
+ * @param req.body.cqLinkStatus The status of the link to CareQuality. (one of "linked",
```

Comment:
hypernit: AFAIK it's supposed to be `Carequality`, no caps on the Q - historically I think the name comes from "care equality" rather than "care quality"